# Problem 10.13

In [2]:
import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [7]:
df = pd.read_csv('wages.csv')
df.columns

Index([u'wage', u'educ', u'exper', u'tenure', u'nonwhite', u'female',
       u'married', u'numdep', u'smsa', u'northcen', u'south', u'west',
       u'construc', u'ndurman', u'trcommpu', u'trade', u'services',
       u'profserv', u'profocc', u'clerocc', u'servocc', u'lwage', u'expersq',
       u'tenursq'],
      dtype='object')

In [4]:
y = df['wage']
X = df['female']
X = sm.add_constant(X)

In [5]:
results = sm.OLS(y,X).fit()

In [6]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                   wage   R-squared:                       0.116
Model:                            OLS   Adj. R-squared:                  0.114
Method:                 Least Squares   F-statistic:                     68.54
Date:                Tue, 05 Dec 2017   Prob (F-statistic):           1.04e-15
Time:                        21:21:17   Log-Likelihood:                -1400.7
No. Observations:                 526   AIC:                             2805.
Df Residuals:                     524   BIC:                             2814.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
const          7.0995      0.210     33.806      0.0

## part i

In [8]:
X2 = df[['female','educ','exper','tenure','married']]
X2 = sm.add_constant(X2)
results2 = sm.OLS(y,X2).fit()
print(results2.summary())

                            OLS Regression Results                            
Dep. Variable:                   wage   R-squared:                       0.368
Model:                            OLS   Adj. R-squared:                  0.362
Method:                 Least Squares   F-statistic:                     60.61
Date:                Tue, 05 Dec 2017   Prob (F-statistic):           1.02e-49
Time:                        21:25:29   Log-Likelihood:                -1312.3
No. Observations:                 526   AIC:                             2637.
Df Residuals:                     520   BIC:                             2662.
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
const         -1.6182      0.723     -2.238      0.0

The coefficient on **female** got smaller in magnitude, but is still a non-trivial effect and the p-value is essentially zero. Adding the other variables in the model seem to increase the explanatory power, as $R^2 = .368$ now which is a reasonable increase (however, as discussed in class, $R^2$ increases as you add variables even if they aren't actually explaining anything more). Also, most of the variables except for **experience** and possibly **married** seem to have a statistically significant effect. Thus, even when controlling for many other factors, **female** still has a strong negative effect.

## part ii

In [9]:
df['married*female'] = df['married']*df['female']
X3 = df[['female','educ','exper','tenure','married','married*female']]
X3 = sm.add_constant(X3)
results2 = sm.OLS(y,X3).fit()
print(results2.summary())

                            OLS Regression Results                            
Dep. Variable:                   wage   R-squared:                       0.392
Model:                            OLS   Adj. R-squared:                  0.385
Method:                 Least Squares   F-statistic:                     55.84
Date:                Tue, 05 Dec 2017   Prob (F-statistic):           3.90e-53
Time:                        21:27:19   Log-Likelihood:                -1302.1
No. Observations:                 526   AIC:                             2618.
Df Residuals:                     519   BIC:                             2648.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [95.0% Conf. Int.]
----------------------------------------------------------------------------------
const             -2.3942      0.730     -3.

Now the coefficient for **female** is much smaller in magnitude, and is statistically insignificant, so we fail to reject the null hypothesis $H_0: \beta_{female} = 0$. However, the coefficients on **married** and __married\*female__ have both become much larger in magnitude and statistically significant. The effect of __married\*female__ is very strong and the effect of __married__ is now more strongly positive. This suggests that much of the wage discrimination falls on married women.